community/kernel/src/main/java/org/neo4j/kernel/IdType.java

```diff
@@ -19,49 +19,83 @@
  */
 package org.neo4j.kernel;
 
-import org.neo4j.unsafe.impl.internal.dragons.FeatureToggles;
+import java.util.Arrays;
+import java.util.List;
 
 /**
  * @deprecated This will be moved to internal packages in the next major release.
  */
 @Deprecated
-public enum IdType
+public class IdType
```

Comment:
Your first note was correct - all of it because of mutable state of idType.
It state will be updated during enterprise db configuration.
In case if we keeping enum to represent id type we need another class to store it state and yet another to group them together in some way - so it's even more complex right?
And i do not want to put mutable structure inside of enum because that is still mutable state